build/types.ts

```diff
@@ -0,0 +1,66 @@
+import { appendFile, writeFile } from "fs/promises";
+import { BlockExplorer, NativeToken } from "../types/handler";
+import { generateChainData } from "../lib/chainlist/utils/fetch";
+
+/**
+ * This produces dynamic types and constants for:
+ * - CHAINS_IDS
+ * - NETWORKS
+ * - NETWORK_FAUCETS
+ * - NETWORK_EXPLORERS
+ * - NETWORK_CURRENCIES
+ * - EXTRA_RPCS
+ */
+
+async function createDynamicTypes() {
+  const data = await generateChainData();
+  const idToNetwork: Record<string, string> = {};
+  const networkToId: Record<string, number> = {};
+  const idToRpc: Record<string, string[]> = {};
+  const idToFaucet: Record<string, string[]> = {};
+  const idToExplorers = {} as Record<string, BlockExplorer[]>;
+  const idToNativeCurrency: Record<string, NativeToken> = {};
+  const extraRpcs: Record<string, string[]> = {};
+
+  for (const chain of data) {
+    let { name, chainId, networkId, rpc, faucets, explorers, nativeCurrency } = chain;
+    name = name.toLowerCase().replace(/\s/g, "-");
+    idToNetwork[chainId] = name;
+    networkToId[name] = networkId;
+    idToRpc[chainId] = rpc;
+    idToFaucet[chainId] = faucets;
+
+    if (explorers && explorers.length > 0) {
+      idToExplorers[chainId] = explorers;
+    }
+
+    if (rpc && rpc.length > 0) {
+      const rpcs = rpc.filter((rpc: { url: string }) => rpc.url);
+      extraRpcs[chainId] = rpcs.map((rpc: { url: string }) => rpc.url);
+    }
+
+    idToNativeCurrency[chainId] = nativeCurrency;
+  }
+
+  // Clear the file
+  await writeFile("types/dynamic.ts", "/* eslint-disable sonarjs/no-duplicate-string */\n\n");
```

Comment:
Instead of ignoring the duplicate string rule, why don't you assign this to a variable and reuse?

Reply:
It's not a feasible approach because the file is 29k in length. The duplicates are things like long and short form crypto names, chain names etc.

P.S a cryptocurrency technically speaking is the native crypto (gas token) of that network and all other "cryptos" are tokens on the network or at least that distinction used to be valid.

Reply:
Colloquially the "coin" is the gas token. Everything else refers to erc20 compliant tokens.

Reply:
Ah I misunderstood, ignoring the rule was for contents of the filename the name but that would have thrown eslint as well.

I know they are all used interchangeably now, it can be easy to misconstrue sometimes